test/suite/test_config08.py

```diff
@@ -0,0 +1,99 @@
+# of this software dedicate any and all copyright interest in the
+# software to the public domain. We make this dedication for the benefit
+# of the public at large and to the detriment of our heirs and
+# successors. We intend this dedication to be an overt act of
+# relinquishment in perpetuity of all present and future rights to this
+# software under copyright law.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+# test_config08.py
+#   Test the configuration that enables/disables dirty tree flushing.
+#
+
+import time, wiredtiger, wttest
+from wtscenario import make_scenarios
+
+class test_config08(wttest.WiredTigerTestCase):
+
+    logging = [
+        ('log_off', dict(log='(enabled=false)')),
+        ('log_on', dict(log='(enabled=true)')),
+    ]
+    fileFlush = [
+        ('flush_off', dict(file_close_sync='false')),
+        ('flush_on', dict(file_close_sync='true')),
+    ]
+
+    scenarios = make_scenarios(logging, fileFlush)
+
+    # This test varies the logging and file flush settings and therefore needs to set up its own
+    # connection. Override the standard methods.
+    def setUpConnectionOpen(self, dir):
+        return None
+    def setUpSessionOpen(self, conn):
+        return None
+    def ConnectionOpen(self, log, file_close_sync):
+        self.home = '.'
+        # Explicitly increase time between checkpoints
+        conn_params = 'create,checkpoint=(wait=60),' + \
+            'log=' + log + ',' + 'file_close_sync=' + file_close_sync
+        try:
+            self.conn = wiredtiger.wiredtiger_open(self.home, conn_params)
+            self.session = self.conn.open_session()
+        except wiredtiger.WiredTigerError as e:
+            print("Failed conn at '%s' with config '%s'" % (dir, conn_params))
+
+    # Create a table with logging setting matching the connection level config.
+    def test_config08(self):
+        self.ConnectionOpen(self.log, self.file_close_sync)
+        table_params = 'key_format=i,value_format=S,log=' + self.log
+        self.uri = 'table:config_test'
+
+        # Create a table with some data
+        self.session.create(self.uri, table_params)
+        c = self.session.open_cursor(self.uri, None)
+        c[0] = 'ABC' * 4096
+        c[1] = 'DEF' * 4096
+        c[2] = 'GHI' * 4096
+        c.close()
+
+        # API calls that require exclusive file handles should return EBUSY if dirty tables flush
+        # and logging are disabled.
+        if self.log == '(enabled=false)' and self.file_close_sync == 'false':
```

Comment:
This changes with the other change I suggest above:
if self.logging == 'false' and ...

Reply:
Fixed.